opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -302,6 +304,8 @@ def add_event(
 
     def add_lazy_event(self, event: trace_api.Event) -> None:
         with self._lock:
+            if not self.is_recording_events():
```

Comment:
As long as we always return `True` in `is_recording_events` I wonder if we need these checks at all yet. And if we want them, maybe they can be outside the lock (I think the sampling decision is made when creating the span already, so this should never change)?

Reply:
Well, I just wanted to put that logic in place regardless the value that is returned by `is_recording_events` (the same that otel-java does right now).

I am not sure if we can move those checks outside the lock, so I preferred to stay in the safe side and leave them inside.